src/string_object.cpp

```diff
@@ -579,6 +579,29 @@ size_t StringObject::char_count(Env *env) const {
     return char_count;
 }
 
+Value StringObject::setbyte(Env *env, Value index_obj, Value value_obj) {
+    assert_not_frozen(env);
+
+    int index = IntegerObject::convert_to_int(env, index_obj);
```

Comment:
Technically `index` could be a larger index, yeah? I think this should use `IntegerObject::convert_to_nat_int_t()`. It may not matter practically, since an `int` could index into a string of 2 Gigs, but it would be nice to match everywhere else we index into strings (which uses either a `size_t` or a `nat_int_t`).

Reply:
Fixed.

I would like to squash commits into a single one before merging. Please let me know if there is anything to change.